### PR TITLE
Initial modules refactor

### PR DESCRIPTION
**Pull request by davepagurek**
This grabs from https://github.com/processing/p5.js/pull/7270 up to the last green check and merges upstream changes in, to hopefully arrive at a mergeable state.

### CODE REVIEW

**Comment by davepagurek** [comment]
@limzykenneth let me know if this looks ok!

File: src/core/structure.js [hunk 1]
```diff
@@ -826,13 +821,7 @@ p5.prototype.push = function() {
...
  */
 p5.prototype.pop = function() {
-  const style = this._styles.pop();
-  if (style) {
-    this._renderer.pop(style.renderer);
-    Object.assign(this, style.props);
-  } else {
-    console.warn('pop() was called without matching push()');
-  }
+  this._renderer.pop();
```
Comment (perminder-17) [inline src/core/structure.js:824]: do we still get a warning on `pop()` was called without matching `push()`?

Comment (limzykenneth) [reply src/core/structure.js:824]: The `p5` instance does not have knowledge about whether any call to `pop()` is valid or not as the states are handled by the renderer. The warning message if needed can be added to `p5.Renderer` base class but I'm not entirely sure if it is useful.

The feature is also not complete (to be completed in #7270), this PR just to lay groundwork for converting the WebGL module to use the new module syntax.

Comment (perminder-17) [reply src/core/structure.js:824]: > ThYe `p5` instance does not have knowledge about whether any call to `pop()` is valid or not as the states are handled by the renderer. The warning message if needed can be added to `p5.Renderer` base class but I'm not entirely sure if it is useful.
> 
> The feature is also not complete (to be completed in #7270), this PR just to lay groundwork for converting the WebGL module to use the new module syntax.

Yes thanks, makes a lot of sense.

**Pull request merged**